src/plugins/messageStorage/sqlite.ts

```diff
@@ -197,11 +210,12 @@ class MessageStorage {
 					}
 				)
 			);
-		});
+		}) as Promise<Message[]>;
 	}
 
-	search(query) {
+	search(query: SearchQuery): Promise<SearchResponse | []> {
 		if (!this.isEnabled) {
+			// TODO: this should return an empty SearchResponse?
```

Comment:
It does here: https://github.com/thelounge/thelounge/blob/9b467af0341c0547bcc8bf02c60af892e3f6f594/src/client.ts#L622

Reply:
added a comment mentioning it. changing the return value can be left for later